packages/vite/src/node/plugins/css.ts

```diff
@@ -557,6 +557,14 @@ export function cssPostPlugin(config: ResolvedConfig): Plugin {
           })
           chunk.viteMetadata.importedCss.add(this.getFileName(fileHandle))
         } else if (!config.build.ssr) {
+          // legacy build and inline css
+
+          // the legacy build should avoid inserting entry CSS modules here, they
+          // will be collected into `chunk.viteMetadata.importedCss` and injected
+          // later by the `'vite:build-html'` plugin into the `index.html`
+          if (chunk.isEntry) {
+            return null
+          }
```

Comment:
I haven't quite caught up with the recent legacy changes, but are we now moving legacy-specific code into core? (just curious 😬)

Reply:
This branch has always been in the `css` plugin. It would be great to move it plugin-legacy but there are some details where we are cheating from the inception of the plugin.

Reply:
maybe no. The previous PR #9507 fix the legacy build css (this if branch) no replace the asset flag(__VITE_ASSET) because the `css-post` later than `asset`. This PR optimized legacy build css generate logic, and this if branch will trigger by legcy build.  😊